backends/tofino/bf-p4c/mau/gateway.cpp

```diff
@@ -160,7 +160,7 @@ const IR::Expression *CanonGatewayExpr::postorder(IR::Operation::Relation *e) {
     Pattern::Match<IR::Expression> e1, e2;
     Pattern::Match<IR::Constant> k1, k2;
     int width = safe_width_bits(e->left->type);
-    if (((e1 & k1) == k2).match(e) || ((e1 & k1) != k2).match(e)) {
+    if (((e1 & k1) == k2).match(e) || ((e1 & k1) == k2).match(e)) {
```

Comment:
This changes what is being matched (from an IR::Neq to an IR::Equ) to completely change the behavior of the code.  It seems just wrong